spring-cloud-deployer-spi-test/src/main/java/org/springframework/cloud/deployer/spi/test/AbstractAppDeployerIntegrationTests.java

```diff
@@ -251,10 +254,84 @@ public void testParameterPassing() {
 		appDeployer().undeploy(deploymentId);
 		assertThat(deploymentId, eventually(hasStatusThat(
 				Matchers.<AppStatus>hasProperty("state", is(unknown))), timeout.maxAttempts, timeout.pause));
+
+		// This second pass makes sure that properties are indeed passed
+
+		properties.put("parameterThatMayNeedEscaping", "notWhatIsExpected");
+		definition = new AppDefinition(randomName(), properties);
+
+		request = new AppDeploymentRequest(definition, integrationTestProcessor(), deploymentProperties);
+
+		log.info("Deploying {}, expecting it to fail...", request.getDefinition().getName());
+
+		deploymentId = appDeployer().deploy(request);
+		timeout = deploymentTimeout();
+		assertThat(deploymentId, eventually(hasStatusThat(
+				Matchers.<AppStatus>hasProperty("state", is(failed))), timeout.maxAttempts, timeout.pause));
+
+		log.info("Undeploying {}...", deploymentId);
+
+		timeout = undeploymentTimeout();
+		appDeployer().undeploy(deploymentId);
+		assertThat(deploymentId, eventually(hasStatusThat(
+				Matchers.<AppStatus>hasProperty("state", is(unknown))), timeout.maxAttempts, timeout.pause));
 	}
 
 	/**
-	 * Tests a simple deploy-undeploy cycle.
+	 * Tests that command line arguments (ordered strings) can be passed to a deployed app,
+	 * including values that typically require special handling.
+	 */
+	@Test
+	public void testCommandLineArgumentsPassing() {
+		Map<String, String> properties = new HashMap<>();
+		AppDefinition definition = new AppDefinition(randomName(), properties);
+		Map<String, String> deploymentProperties = new HashMap<>();
+
+		List<String> cmdLineArgs = Arrays.asList("--commandLineArgValueThatMayNeedEscaping=d" + DeployerIntegrationTestProperties.FUNNY_CHARACTERS);
```

Comment:
Hmmm, this is not good. I added it to test that it indeed would fail, but must have forgotten to remove it. At first I was surprised that an arg that contained a space could be passed as is, so doubted of the quality of the test. I then must have been carried away.